providers/openebs/cspc_csi_upgrade/test.yml

```diff
@@ -0,0 +1,509 @@
+                  url: "https://raw.githubusercontent.com/openebs/cstor-csi/master/deploy/csi-operator.yaml"
+                  dest: "{{ playbook_dir }}/csi-operator.yaml"
+                  force: yes
+                register: status
+                until:  "'OK' in status.msg"
+                delay: 5
+                retries: 3
+
+              - name: Update the desired image tag for openebs-plugin
+                replace:
+                  path: "{{ playbook_dir }}/csi-operator.yaml"
+                  regexp: "cstor-csi-driver:ci"
+                  replace: "cstor-csi-driver:{{ to_version }}"
+
+              - name: Update the label tag for openebs-plugin
+                replace:
+                  path: "{{ playbook_dir }}/csi-operator.yaml"
+                  regexp: "openebs.io/version: dev"
+                  replace: "openebs.io/version: {{ to_version }}"
+
+              - name: Deploy CSI Driver if OS is either centos or ubuntu-16.04
+                shell: >
+                  kubectl apply -f csi-operator.yaml
+                args:
+                  executable: /bin/bash
+                register: deploy_status
+                failed_when: "deploy_status.rc != 0"
+
+              when: node_os == "ubuntu-16.04" or node_os == "centos"
+
+            - block:
+
+              - name: Downloading the csi-operator file
+                get_url:
+                  url: "https://raw.githubusercontent.com/openebs/cstor-csi/master/deploy/csi-operator-ubuntu-18.04.yaml"
+                  dest: "{{ playbook_dir }}/csi-operator-ubuntu-18.04.yaml"
+                  force: yes
+                register: status
+                until:  "'OK' in status.msg"
+                delay: 5
+                retries: 3
+
+              - name: Update the desired image tag for openebs-plugin
+                replace:
+                  path: "{{ playbook_dir }}/csi-operator-ubuntu-18.04.yaml"
+                  regexp: "cstor-csi-driver:ci"
+                  replace: "cstor-csi-driver:{{ to_version }}"
+
+              - name: Update the label tag for openebs-plugin
+                replace:
+                  path: "{{ playbook_dir }}/csi-operator-ubuntu-18.04.yaml"
+                  regexp: "openebs.io/version: dev"
+                  replace: "openebs.io/version: {{ to_version }}"
+
+              - name: Deploy CSI Driver if OS is either centos or ubuntu-16.04
+                shell: >
+                  kubectl apply -f csi-operator-ubuntu-18.04.yaml
+                args:
+                  executable: /bin/bash
+                register: deploy_status
+                failed_when: "deploy_status.rc != 0"
+
+              when: node_os == "ubuntu-18.04"
+
+              # By default, volume mount is disabled. When it is enabled, the below patching tasks should be ignored.
+
+            - name: Identify the patch to be invoked
+              template:
+                src: patch.j2
+                dest: patch.yml
```

Comment:
why do we patch in upgrade job? It would already be done by csi-provisioner

Reply:
In case of provisioning, if the auto remount is not enabled then we need this. so only added it here after upgrade the csi provisioner

Reply:
it is not necessary. we are doing this as part of csi provisioned. It should be enabled in the file itself for this release. you can remove this